test/index.js

```diff
@@ -314,6 +314,48 @@ describe('instantiation', function() {
     assert(firestore instanceof Firestore);
   });
 
+  it('merges settings', function() {
+    let firestore = new Firestore({
+      projectId: PROJECT_ID,
+    });
+
+    firestore.settings({
+      timestampsInSnapshots: true,
+    });
+
+    assert.equal(firestore._initalizationOptions.projectId, PROJECT_ID);
+    assert.equal(firestore._initalizationOptions.timestampsInSnapshots, true);
+  });
+
+  it('can only call settings() once', function() {
+    let firestore = new Firestore();
+    firestore.settings({});
+
+    assert.throws(
+        () => firestore.settings({}),
+        /Firestore.settings\(\) has already be called. You can only call settings\(\) once, and only before calling any other methods on a Firestore object./);
+  });
+
+  it('cannot change settings after client initialized', function() {
+    return createInstance().then(firestore => {
+      assert.throws(
+          () => firestore.settings({}),
+          /Firestore has already been started and its settings can no longer be changed. You can only call settings\(\) before calling any other methods on a Firestore object./);
+    });
+  });
+
+  it('validates project ID is string', function() {
```

Comment:
Add similar test for validating the value of `timestampsInSnapshots`.

Reply:
Done (also added the missing validation in the constructor).